src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/controller/ListingController.java

```diff
@@ -50,10 +51,44 @@ public List<ListingGetDto> getAllListings() {
     @ResponseStatus(HttpStatus.CREATED)
     @ResponseBody
     public ListingGetDto createListing(@RequestBody ListingPostDto listingPostDto) {
-        Listing listingInput = ListingMapper.INSTANCE.listingPostDtoToListing(listingPostDto);
+        //Listing listingInput = ListingMapper.INSTANCE.listingPostDtoToListing(listingPostDto);
+
+        Location address = null;
+
+        if (listingPostDto.getAddress() != null) {
+            address = new Location(
+                    listingPostDto.getName(),
+                    listingPostDto.getName(),
+                    listingPostDto.getAddress().getStreet(),
+                    listingPostDto.getAddress().getHouseNumber(),
+                    listingPostDto.getAddress().getApartmentNumber(),
+                    listingPostDto.getAddress().getZipCode(),
+                    listingPostDto.getAddress().getCity(),
+                    listingPostDto.getAddress().getCountry()
+            );
+        }
 
         // TODO: User Authentication required to create listings?
-        Listing createdListing = listingService.createListing(listingInput);
+        //Listing createdListing = listingService.createListing(listingInput, address);
+        Listing createdListing = listingService.createListing(new Listing(
```

Comment:
if mapper using Annotation does not work, consider removing the ListingMapper interface, create a ListingMapper class and create static methods to map DTO to listing and listing to DTO so we have less duplicate code